argilla/src/argilla/datasets/_export/_hub.py

```diff
@@ -0,0 +1,199 @@
+
+        Parameters:
+            repo_id: the ID of the Hugging Face Hub repo to load the `Dataset` from.
+            workspace: the workspace to load the `Dataset` into. If not provided, the default workspace will be used.
+            client: the client to use to load the `Dataset`. If not provided, the default client will be used.
+            with_records: whether to load the records from the Hugging Face dataset. Defaults to `True`.
+            *args: the args to pass to `datasets.Dataset.load_from_hub`.
+            **kwargs: the kwargs to pass to `datasets.Dataset.load_from_hub`.
+
+        Returns:
+            A `Dataset` loaded from the Hugging Face Hub.
+        """
+        from datasets import DatasetDict, load_dataset, Dataset
+        from huggingface_hub import snapshot_download
+
+        folder_path = snapshot_download(  # download both files in parallel
+            repo_id=repo_id,
+            repo_type="dataset",
+            allow_patterns=cls._DEFAULT_CONFIGURATION_FILES,
+            token=kwargs.get("token"),
+        )
+
+        dataset = cls.from_disk(path=folder_path, target_workspace=workspace, client=client)
+
+        hf_dataset: Dataset = load_dataset(path=repo_id, *args, **kwargs)  # type: ignore
+        if isinstance(hf_dataset, DatasetDict) and "split" not in kwargs:
+            if len(hf_dataset.keys()) > 1:
+                raise ValueError(
+                    "Only one dataset can be loaded at a time, use `split` to select a split, available splits"
+                    f" are: {', '.join(hf_dataset.keys())}."
+                )
+            hf_dataset: Dataset = hf_dataset[list(hf_dataset.keys())[0]]
+
+        if with_records:
+            cls._log_dataset_records(hf_dataset=hf_dataset, dataset=dataset)
+        return dataset
+
+    @staticmethod
+    def _log_dataset_records(hf_dataset: "HFDataset", dataset: "Dataset"):
+        """This method extracts the responses from a Hugging Face dataset and returns a list of `Record` objects"""
+
+        # Identify columns that colunms that contain responses
+        responses_columns = [col for col in hf_dataset.column_names if ".responses" in col]
+        response_questions = defaultdict(dict)
+        user_ids = {}
+        for col in responses_columns:
+            question_name = col.split(".")[0]
+            if col.endswith("users"):
+                response_questions[question_name]["users"] = hf_dataset[col]
+                user_ids.update({UUID(user_id): UUID(user_id) for user_id in set(sum(hf_dataset[col], []))})
+            elif col.endswith("responses"):
+                response_questions[question_name]["responses"] = hf_dataset[col]
+            elif col.endswith("status"):
+                response_questions[question_name]["status"] = hf_dataset[col]
+
+        # Check if all user ids are known to this Argilla client
+        known_users_ids = [user.id for user in dataset._client.users._api.list()]
```

Comment:
using public client api?
```suggestion
        known_users_ids = [user.id for user in dataset._client.users]
```